packages/amplify_api/android/src/test/kotlin/com/amazonaws/amplify/amplify_api/GraphQLApiUnitTests.kt

```diff
@@ -0,0 +1,299 @@
+                any<Consumer<GraphQLResponse<String>>>(),
+                any())
+
+
+        flutterPlugin.onMethodCall(
+                MethodCall("mutate", testRequest),
+                mockResult
+        )
+
+        verify(mockResult, times(1)).success(mapOf(
+                "data" to "mutate result",
+                "errors" to listOf<String>()
+        ))
+
+    }
+
+    @Test
+    fun test_mutate_malformed_request_casting_error() {
+        val testRequest = HashMap<String, Any>()
+
+        flutterPlugin.onMethodCall(
+                MethodCall("mutate", testRequest),
+                mockResult
+        )
+
+        verify(mockResult, times(1)).error(
+                matches("AmplifyException"),
+                matches(FlutterApiErrorMessage.AMPLIFY_REQUEST_MALFORMED.toString()),
+                any()
+        )
+    }
+
+    @Test
+    fun test_mutate_api_error() {
+        val apiException = ApiException("AmplifyException", ApiException.REPORT_BUG_TO_AWS_SUGGESTION)
+        val testRequest = HashMap<String, Any>()
+
+        testRequest["document"] = ("mutation MyMutation(\$name: String!) {"
+                + "createBlog(input: {name: \$name}) {"
+                + "id"
+                + "name"
+                + "createdAt"
+                + "}"
+                + "}")
+
+        testRequest["variables"] = mapOf(
+                "name" to "Test App Blog"
+        )
+
+        doAnswer { invocation ->
+            Assert.assertEquals(
+                    SimpleGraphQLRequest<String>(
+                            testRequest["document"] as String,
+                            testRequest["variables"] as Map<String, Any>,
+                            String::class.java,
+                            GsonVariablesSerializer()
+                    ),
+                    invocation.arguments[0]
+            )
+            (invocation.arguments[2] as Consumer<ApiException>).accept(
+                    apiException
+            )
+            mockGraphQLOperation
+        }.`when`(mockApi).mutate(
+                any<GraphQLRequest<String>>(),
+                any(),
+                any())
+
+
+        flutterPlugin.onMethodCall(
+                MethodCall("mutate", testRequest),
+                mockResult
+        )
+
+        verify(mockResult, times(1)).error(
+                "AmplifyException",
+                FlutterApiErrorMessage.AMPLIFY_API_MUTATE_FAILED.toString(),
+                mapOf(
+                        "PLATFORM_EXCEPTIONS" to
+                                mapOf(
+                                        "platform" to "Android",
+                                        "localizedErrorMessage" to "AmplifyException",
+                                        "recoverySuggestion" to ApiException.REPORT_BUG_TO_AWS_SUGGESTION,
+                                        "errorString" to apiException.toString()))
+        )
+
+    }
+
+    private fun setFinalStatic(field: Field, newValue: Any?) {
```

Comment:
Once you update your flutter plugin to call `ApiCategoryBehavior` instead of `Amplify.API` directly, you will no longer need to do this. You can just use a `mock(ApiPlugin)` and then arrange behaviors on it. In your test, you'll pass the mock plugin down into your Flutter wrapper plugin.

Reply:
Nice! Will address as part of the test refactor PR.